js/src/block-editor/hooks/useMedia.js

```diff
@@ -58,6 +59,13 @@ const useMedia = ( fieldValue, onChange, allowedTypes ) => {
 		return select( 'core' ).getMedia( fieldValue );
 	} );
 
+	/* @type {function|undefined} */
+	const mediaUpload = useSelect( ( select ) => {
```

Comment:
Mainly copied from the [Core Image block](https://github.com/WordPress/gutenberg/blob/dffade6ba05c07cab78ffe3432cb5ef0aaae2364/packages/block-library/src/image/edit.js#L124-L127
)